src/v/security/oidc_authenticator.cc

```diff
@@ -134,4 +134,55 @@ authenticator::authenticate(std::string_view bearer_token) {
     return _impl->authenticate(bearer_token);
 }
 
+std::ostream& operator<<(std::ostream& os, sasl_authenticator::state const s) {
+    using state = sasl_authenticator::state;
+    switch (s) {
+    case state::init:
+        return os << "init";
+    case state::complete:
+        return os << "complete";
+    case state::failed:
+        return os << "failed";
+    }
+}
+
+sasl_authenticator::sasl_authenticator(service& service)
+  : _authenticator{service} {}
+
+sasl_authenticator::~sasl_authenticator() = default;
+
+ss::future<result<bytes>> sasl_authenticator::authenticate(bytes auth_bytes) {
+    if (_state != state::init) {
+        vlog(seclog.debug, "invalid oidc state: {}", _state);
+        co_return security::errc::invalid_oidc_state;
+    }
+
+    auto auth_str = std::string_view(
+      reinterpret_cast<char*>(auth_bytes.data()), auth_bytes.size());
+
+    constexpr std::string_view sasl_header{"n,,\1auth=Bearer "};
+    if (!auth_str.starts_with(sasl_header)) {
+        vlog(seclog.debug, "invalid sasl_header");
+        co_return security::errc::invalid_credentials;
+    }
+    auth_str = auth_str.substr(sasl_header.length());
+    if (!auth_str.ends_with("\1\1")) {
+        vlog(seclog.debug, "invalid sasl_header");
+        co_return security::errc::invalid_credentials;
+    }
+    auth_str = auth_str.substr(0, auth_str.length() - 2);
+    vlog(seclog.trace, "auth_bytes: {}", auth_str);
```

Comment:
question: any security issues with logging this?

Reply:
> question: any security issues with logging this?

Not really, but we don't explicitly check against replay attacks yet. I'll remove it.

Reply:
Removed